src/libYARP_robotinterface/src/yarp/robotinterface/Robot.cpp

```diff
@@ -572,13 +672,13 @@ bool yarp::robotinterface::Robot::enterPhase(yarp::robotinterface::ActionPhase p
                 }
                 break;
             case ActionTypeCalibrate:
-                if(!mPriv->calibrate(device, action.params())) {
+                if(!mPriv->calibrate(device, action.params(), externalDevicesList)) {
```

Comment:
I assume that calibrate, attach and park are allowed to have an external device list, because these require an interaction between devices that might be external, while and not configure, abort and detach don't, am I right?

Perhaps also `custom` should have this list? (even though, I think it might just be a placeholder, that was never implemented)

Reply:
> I assume that calibrate, attach and park are allowed to have an external device list, because these require an interaction between devices that might be external, while and not configure, abort and detach don't, am I right?

Exactly. I  just added the parameter in the methods that where actually using internally `hasDevice`  or `findDevice` .